src/agent/vsoworker.ts

```diff
@@ -109,7 +109,7 @@ export function run(msg: cm.IWorkerMessage, consoleOutput: boolean,
         trace.state('msg:', msg);
 
         var agentUrl = hostContext.config.settings.serverUrl;
-        var taskUrl = job.environment.variables[cm.vars.systemTfCollectionUri];
+        var taskUrl = job.environment.systemConnection.url;
```

Comment:
line 38 will never get executed from here on. this variable will be set by the service. 
Post s89 deployments systemconnection.url and system.teamfoundationCollectionUri will have different values. systemConnection.Url will have the service specific url and the variable will point to tfs collection url always..
Did I miss something really basic here.. ? :)